frontend/components/canvas.tsx

```diff
@@ -1,75 +1,85 @@
-"use client";
+'use client'
 
-import { useDraw } from "@/hooks/useDraw";
-import { useState } from "react";
-import { CompactPicker } from "react-color";
+import { FC, useEffect, useState } from 'react'
+import { useDraw } from '../hooks/useDraw'
+import { ChromePicker } from 'react-color'
 
-export default function Canvas() {
-  const [color, setColor] = useState<string>('#000');
-  const [width, setWidth] = useState<number>(5);
-  const { canvasRef, onMouseDown, clear } = useDraw(drawLine);
+import { io } from 'socket.io-client'
+import { drawLine } from '../util/drawLine'
+const socket = io('http://localhost:3001')
 
-  function drawLine({ prevPoint, currentPoint, context }: Draw) {
-    const { x: currX, y: currY } = currentPoint;
-    const lineColor = color;
-    const lineWidth = width;
+interface pageProps {}
 
-    let startPoint = prevPoint ?? currentPoint;
-    context.beginPath();
-    context.lineWidth = lineWidth;
-    context.strokeStyle = lineColor;
-    context.moveTo(startPoint.x, startPoint.y);
-    context.lineTo(currX, currY);
-    context.stroke();
+type DrawLineProps = {
+  prevPoint: Point | null
+  currentPoint: Point
+  color: string
+}
+
+const page: FC<pageProps> = () => {
+  const [color, setColor] = useState<string>('#000')
+  const { canvasRef, onMouseDown, clear } = useDraw(createLine)
+
+  useEffect(() => {
+    const ctx = canvasRef.current?.getContext('2d')
+
+    socket.emit('client-ready')
+
+    socket.on('get-canvas-state', () => {
+      if (!canvasRef.current?.toDataURL()) return
+      console.log('sending canvas state')
+      socket.emit('canvas-state', canvasRef.current.toDataURL())
```

Comment:
Need to create a new event handler on the backend to accept this "canvas-state" event in order for the server to store the canvas state to give to other clients later

Update `backend\src\socket.ts`, something along the lines of
```
socket.on("canvas-state", (state) => {
      const gameId = socket.data.gameId;
      if (gameId !== null) {
        <store canvas state somewhere>
      }
    });
```

and update `ClientToServerEvents` so our typing expects this event name

Reply:
I would probably create a new function in `game.ts` called `storeGameCanvas` that takes in a `gameId` and `state` and store the state on the `game` object for that `gameId`